src/core/playground/wrapped.ts

```diff
@@ -0,0 +1,79 @@
+import type { Deserializer, Serializer } from "../../bcs";
+import { HexInput, SigningScheme as AuthenticationKeyScheme } from "../../types";
+import { AuthenticationKey } from "./authenticationKey";
+import { bcsToBytes } from "./bcs";
+import { Ed25519PublicKey, Ed25519Signature } from "./ed25519";
+import type { PublicKey } from "./interfaces";
+import { SignatureScheme } from "./scheme";
+import { Secp256k1PublicKey, Secp256k1Signature } from "./secp256k1";
+
+export type AllowedSignatures = Ed25519Signature | Secp256k1Signature;
+
+export class AnySignature<TSignature extends AllowedSignatures = AllowedSignatures> {
+  constructor(public readonly signature: TSignature) {}
+
+  serialize(serializer: Serializer): void {
+    if (this.signature instanceof Ed25519Signature) {
+      serializer.serializeU32AsUleb128(SignatureScheme.Ed25519);
+    } else if (this.signature instanceof Secp256k1Signature) {
+      serializer.serializeU32AsUleb128(SignatureScheme.Secp256k1);
+    } else {
+      throw new Error("Unknown signature type");
+    }
+    this.signature.serialize(serializer);
+  }
+
+  static deserialize(deserializer: Deserializer): AnySignature {
+    const scheme = deserializer.deserializeUleb128AsU32();
+    switch (scheme) {
+      case SignatureScheme.Ed25519:
+        return new AnySignature(Ed25519Signature.deserialize(deserializer));
+      case SignatureScheme.Secp256k1:
+        return new AnySignature(Secp256k1Signature.deserialize(deserializer));
+      default:
+        throw new Error(`Unknown signature scheme ${scheme}`);
+    }
+  }
+}
+
+export class AnyPublicKey<
+  TSignature extends AllowedSignatures = AllowedSignatures,
+  TPublicKey extends PublicKey<TSignature> = PublicKey<TSignature>,
+> implements PublicKey<AnySignature<TSignature>>
+{
+  constructor(public readonly publicKey: TPublicKey) {}
```

Comment:
can we please define `public readonly publicKey: TPublicKey` in the class and not in the  constructor signature